leveldb-server/src/test/db.match.spec.ts

```diff
@@ -0,0 +1,175 @@
+import test, { Macro } from 'ava';
+import { testing } from '../db';
+import { Filter } from '@binaris/shift-interfaces-koa-server/interfaces';
+
+const { match } = testing;
+
+type Operator = 'eq' | 'ne' | 'gt' | 'lt' | 'gte' | 'lte';
+type Scalar = string | number | boolean;
+
+const valueMatches: Macro<[any, Operator, Scalar]> = (t, testValue, operator, matchValue) => {
+  // TypeScript type system not strong enough to deduce this always
+  // creates a legal filter, so just cast.
+  const filter = { operator, path: ['value'], value: matchValue } as Filter;
+  t.true(match({ key: 'abc', value: testValue }, filter));
+};
+valueMatches.title = (providedTitle = '', testValue, operator, matchValue) =>
+  `${providedTitle} ${operator}(${JSON.stringify(testValue)}, ${JSON.stringify(matchValue)})`;
+
+const valueDoesntMatch: Macro<[any, Operator, Scalar]> = (t, testValue, operator, matchValue) => {
+  // TypeScript type system not strong enough to deduce this always
+  // creates a legal filter, so just cast.
+  const filter = { operator, path: ['value'], value: matchValue } as Filter;
+  t.false(match({ key: 'abc', value: testValue }, filter));
+};
+valueDoesntMatch.title = (providedTitle = '', testValue, operator, matchValue) =>
+  `${providedTitle} !${operator}(${JSON.stringify(testValue)}, ${JSON.stringify(matchValue)})`;
+
+const keyMatches: Macro<[string, Operator, string]> = (t, testKey, operator, matchKey) => {
+  // TypeScript type system not strong enough to deduce this always
+  // creates a legal filter, so just cast.
+  const filter = { operator, path: ['key'], value: matchKey } as Filter;
+  t.true(match({ key: testKey, value: 17 }, filter));
+};
+keyMatches.title = (providedTitle = '', testKey, operator, matchKey) =>
+  `${providedTitle} ${operator}(key ${JSON.stringify(testKey)}, key ${JSON.stringify(matchKey)})`;
+
+const keyDoesntMatch: Macro<[string, Operator, string]> = (t, testKey, operator, matchKey) => {
+  // TypeScript type system not strong enough to deduce this always
+  // creates a legal filter, so just cast.
+  const filter = { operator, path: ['key'], value: matchKey } as Filter;
+  t.false(match({ key: testKey, value: 17 }, filter));
+};
+keyDoesntMatch.title = (providedTitle = '', testKey, operator, matchKey) =>
+  `${providedTitle} !${operator}(key ${JSON.stringify(testKey)}, key ${JSON.stringify(matchKey)})`;
+
+test('eq checks for strict equality', valueMatches, 3, 'eq', 3);
+test('eq checks for strict equality', valueDoesntMatch, 3, 'eq', 2);
+test('eq checks for strict equality', valueDoesntMatch, 3, 'eq', '3');
```

Comment:
nice!
I'd test null and 'null', undefined and 'undefined' too since it's interesting

Reply:
Same tests as we had for `db`.

We cannot test `undefined` because that's not a DB value (not JSONable).  And this test doesn't check every type combination; rather, it asserts that we are using `===` and not `==`.  Adding more cases will not make it stricter.